File: backend/models/Dieta.js
const mongoose = require('mongoose');

const dietaSchema = new mongoose.Schema({
  usuario: { type: mongoose.Schema.Types.ObjectId, ref: 'Usuario', required: true },
  fecha: { type: Date, required: true },
  platos: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Plato' }], // Referencia a los platos
}, { timestamps: true });

module.exports = mongoose.model('Dieta', dietaSchema);

File: frontend/lib/models/dieta.dart
import 'plato.dart';

class Dieta {
  final String id;
  final String usuario;
  final DateTime fecha;
  final List<Plato> platos;

  Dieta({
    required this.id,
    required this.usuario,
    required this.fecha,
    required this.platos,
  });

  factory Dieta.fromJson(Map<String, dynamic> json) {
    return Dieta(
      id: json['_id'] ?? '',
      usuario: json['usuario'] is Map
          ? json['usuario']['_id'] ?? ''
          : json['usuario'] ?? '',
      fecha: DateTime.tryParse(json['fecha'] ?? '') ?? DateTime.now(),
      platos: (json['platos'] as List? ?? [])
          .map((e) => Plato.fromJson(e))
          .toList(),
    );
  }
}
